File: spec/lua/test_params_def.lua
local luaunit = require("luaunit")

require("params_def")
require("kaitaistruct")

TestParamsDef = {}

function TestParamsDef:test_params_def()
    local _io = KaitaiStream(io.open("src/term_strz.bin", "rb"))
    local r = ParamsDef(5, true, _io, nil, nil)

    luaunit.assertEquals(r.buf, "foo|b")
    luaunit.assertEquals(r.trailer, 0x61)

    _io:close()    
end

File: spec/javascript/test_params_def.js
var assert = require('assert');
var testHelperStream = require('testHelperStream');

testHelperStream('ParamsDef', 'src/term_strz.bin', function(io, ParamsDef) {
  var r = new ParamsDef(io, null, null, 5, true);

  assert.strictEqual(r.buf, 'foo|b');
  assert.strictEqual(r.trailer, 0x61);
});
